addons/calendar_sms/models/calendar_event.py

```diff
@@ -13,15 +13,19 @@ def _sms_get_default_partners(self):
         """
         return self.mapped('attendee_ids').filtered(lambda att: att.state != 'declined' and att.partner_id.phone_sanitized).mapped('partner_id')
 
-    def _do_sms_reminder(self, alarm):
+    def _do_sms_reminder(self, alarms):
         """ Send an SMS text reminder to attendees that haven't declined the event """
         for event in self:
-            event._message_sms_with_template(
-                template=alarm.sms_template_id,
-                template_fallback=_("Event reminder: %(name)s, %(time)s.", name=event.name, time=event.display_time),
-                partner_ids=self._sms_get_default_partners().ids,
-                put_in_queue=False
-            )
+            for alarm in alarms:
```

Comment:
I'm curious, did this cause issues? Can you actually receive multiple alarms here? In other words, should be backport a fix?

Reply:
Well, after having a look to refresh all this, here is what I can say:

Indeed, "alarm" could contain more than one alarm at the moment. I've done the change because it was confusing and could in some very specific cases lead to an error. See below.

We would have in "alarm" all the sms alarms that the cron should trigger, hence satisfying both conditions
```
trigger_time < now
trigger_time > last cron call
```
where `trigger_time = event start - alarm duration.`

For instance. I have a meeting at 4pm. One sms alarm is 1 hour before (3pm) and one 3 hours before (1pm).
Default value for the cron interval is 1 day. So if last call was yesterday at 3h15, next call, today at 3h15 will trigger both alarms.

But in addition to that, in order to be an issue, we should also have different sms_template_ids set on those alarms, otherwise alarm.sms_template_id will still be a singleton, all good in the previous version.

So yes this can lead to a bug, but it would require two templates, set on two sms alarms, in a time configuration with close reminders set on the event.

**Actually, I also see that cron triggers are set when an event is created or its alarms modified (on the same conditions). So If I get it right, it would also need more than one alarm of the same duration to create an error. Cron triggers have a 1minute precision**

What do you think? Useful change? Unnecessary?

Reply:
Fair enough, seems a bit extreme.

We may backport if clients complain but otherwise let's keep the changes here.

Thanks for investigating!